### PR TITLE
Fixes #264, #191, #194, #271

### LINKED ISSUE #264: Add new shapes especially for direction indication in CanvasGrid
Presently Rectangle and Circle shapes are only supported in CanvasGrid. Sometime, there is a requirement of showing the heading (direction s.t. N/S/E/W) of the agent as well.Some more shapes might be introduced for this very purpose. It would be nice to have support of images as well.

### PR DESCRIPTION
**Pull request by jeetu7**
I tried doing some fixes. Hoping that everything works fine.
Fixes #264, #191, #194, #271

### CODE REVIEW

**Comment by jackiekazil** [comment]
I will comb through this tomorrow. I will try to do it early since I am at Scipy and sprints start tomorrow.

**Comment by jeetu7** [comment]
Seems I have pushed the PR at the wrong time. You might be having a crammed schedule.

**Comment by jackiekazil** [comment]
@jeetu7 no worries. :-) I am going to try to get someone to review during Scipy.

**Comment by jackiekazil** [comment]
Related: https://groups.google.com/forum/#!topic/projectmesa/E3j2jWlM34w

File: tests/test_grid.py [hunk 1]
```diff
@@ -6,10 +6,19 @@
...
 # Initial agent positions for testing
-#              X ---- >
-TEST_GRID = [[0, 1, 0, 1, 0],   # Y
-             [0, 1, 1, 0, 0],   # |
-             [0, 0, 0, 1, 0]]   # V
+#
+# --- visual aid ----
+#   0 0 0
+#   1 1 0
+#   0 1 0
```
Comment (jeetu7) [inline tests/test_grid.py]: visual-aid part in the comment is the grid to be perceived visually. Imagine it to be the real world grid where-in x direction has 3 units and y direction has 5 units.The x direction goes from left to right while the y direction goes from bottom to top.  On the other hand TEST_GRID part (which is out of comment scope) is the actual representation in python list of list so as to get the indexing equivalent (one-to-one mapping)  of real word grid..

File: mesa/space.py [hunk 2]
```diff
@@ -71,11 +71,11 @@ def __init__(self, height, width, torus):
 
         self.grid = []
 
-        for y in range(self.height):
-            row = []
-            for x in range(self.width):
-                row.append(self.default_val())
-            self.grid.append(row)
+        for x in range(self.width):
+            col = []
```
Comment (jeetu7) [inline mesa/space.py]: Look at test_grid.py in the tests directory for understanding the structure of representation. Also see the note which should be visible in the PR conversation history as well. That comment with one example could be added into the documentation part of this file for understanding of future developers.

**Comment by jackiekazil** [comment]
@jeetu7 can you update Basic to follow the format in #223?

**Comment by jeetu7** [comment]
@jackiekazil completed example restructuring.

**Comment by jackiekazil** [comment]
Thank you.... I am continuing to review now.

**Comment by jackiekazil** [comment]
:-( My bad. @jeetu7, can you resolve the merge conflict? (I think it has to do with the readme. I am going through the rest of the code at the moment.

**Comment by jackiekazil** [comment]
Review of this ticket: 
- [ ] #264 
- [ ] #191 
- [ ] #194 
- [ ] #271

**Comment by jeetu7** [comment]
@jackiekazil  I have merged the conflicts in readme. But some formatting seems to be lost. Just check "Using Pre-Built Modules" section in the readme file. The same issue exists in the readme file exists of present master mesa repository. It looks to me that this issue is because of how github markdown  (or markdown in general) works and how our editors handles trailing whitespace (mine vim setting trims it). Something similar might have happened when you edited readme as well. I would suggest pull this PR and then ask  somebody familiar with github markdown at scipy for help.

**Comment by jackiekazil** [comment]
Closing this PR. #285 replaces it. 
It looks like everything is in good order in this PR. 
I created the other to fix some spacing issues, then I realize that I was fixing issues that were unrelated to this PR, so I stopped.

**Comment by jackiekazil** [comment]
@jeetu7 the work in this is really great and really helpful!! Thank you!!! 
It helps to have smaller PRs and to give a heads up when a larger one is going to be coming in. <-- That would have been difficult to do with this with multiple PRs. This was a really complicated issue with many interlacing pieces, which is almost impossible to do since it touches so many pieced of code. The only thing that I could have possibly been separated out was #271.

**Pull request closed without merging**